README.md

```diff
@@ -3,741 +3,470 @@
+Allocators can be purely onchain abstractions, or can involve hybrid (onchain + offchain) mechanics as part of their authorization procedure. Should an allocator erroneously or maliciously fail to authorize the use of an unallocated resource lock balance, the depositor can initiate a **forced withdrawal** for the lock in question; after waiting for the reset period indicated when depositing into the lock, they can withdraw their underlying balance at will _without_ the allocator's explicit permission.
 
-## Setup
-```
-# install foundry if needed
-$ curl -L https://foundry.paradigm.xyz | bash
+Sponsors can also optionally assign an **emissary** to act as a fallback signer for authorizing claims against their compacts. This is particularly helpful for smart contract accounts or other scenarios where signing keys might change.
 
-# clone repo
-$ git clone git@github.com:Uniswap/the-compact.git && cd the-compact
-
-# install dependencies & libraries
-$ forge install
-
-# run basic tests & gas snapshots
-$ forge test
+The Compact effectively "activates" any deposited tokens to be instantly spent or swapped across arbitrary, asynchronous environments as long as:
+ - Claimants are confident that the allocator is sound and will not leave the resource lock underallocated.
+ - Sponsors are confident that the allocator will not unduly censor fully allocated requests.
+ - Sponsors are confident that the arbiter is sound and will not process claims where the conditions were not successfully met.
+ - Claimants are confident that the arbiter is sound and will not *fail* to process claims where the conditions *were* successfully met.
+
+## Key Concepts
+
+### Resource Locks
+Resource locks are the fundamental building blocks of The Compact protocol. They are created when a depositor places tokens (either native tokens or ERC20 tokens) into The Compact. Each resource lock has four key properties:
+
+1.  The **underlying token** held in the resource lock.
+2.  The **allocator** tasked with cosigning on claims against the resource locks (see [Allocators](#allocators)).
+3.  The **scope** of the resource lock (either spendable on any chain or limited to a single chain).
+4.  The **reset period** for forcibly exiting the lock (see [Forced Withdrawals](#forced-withdrawals)) and for emissary reassignment timelocks (see [Emissaries](#emissaries)).
+
+Each unique combination of these four properties is represented by a fungible ERC6909 tokenID. The owner of these ERC6909 tokens can act as a sponsor and create compacts.
+
+The `scope`, `resetPeriod`, and the `allocatorId` (obtained when an allocator is registered) are packed into a `bytes12 lockTag`. A resource lock's specific ID (the ERC6909 `tokenId`) is a hash of this `lockTag` and the `token` address. This `lockTag` is used throughout various interfaces to succinctly identify the parameters of a lock.
```

Comment:
Not a hash; the two values concatenated into a single value and represented as a uint256 for ERC6909 compatibility

Reply:
Ha yeah whoops